### ISSUE TITLE
[docs] Add architecture overview 

### ISSUE THREAD

**Issue by fredguth** [a-type/verdant#248]
As a developer, one of the first things I was expecting to see in the documentation was an architecture overview. Preferably, with a very clear image.  I would like to understand not only how **Verdant** works but also, where each element of the architecture is in your repo.

**Comment by a-type** [comment]
Thanks for taking a look! I'm not totally confident I understand what you expect from architecture-- something high level like "this is the server, this is the client"? Or a deep dive into, say, how the client stores and queries data, or how sync works?

But I agree the README is pretty sparse and the docs aren't too much better. Since this has mostly been a personal project and rapidly evolving, I've been wary of investing too much into documenting what may change dramatically. But I suppose it's about time, since I haven't fundamentally changed how anything works in a good while.

I added some more detail about usage and a high-level overview of internals to the README. If there's anything more you'd like to know specifically, let me know so I can expand it. I've spent so long in this codebase it's hard for me to see it like a newcomer would.
